airbyte-cdk/python/airbyte_cdk/sources/declarative/declarative_stream.py

```diff
@@ -111,9 +112,15 @@ def read_records(
         cursor_field: List[str] = None,
         stream_slice: Mapping[str, Any] = None,
         stream_state: Mapping[str, Any] = None,
-    ) -> Iterable[Mapping[str, Any]]:
-        for record in self.retriever.read_records(sync_mode, cursor_field, stream_slice, stream_state):
-            yield self._apply_transformations(record, self.config, stream_slice)
+    ) -> Iterable[StreamData]:
+        # Set the log level on the retriever if it has a logger
+        if hasattr(self.logger, "level") and hasattr(self.retriever, "logger"):
```

Comment:
as per offline discussion, we can refactor the logger on the retriever to share the same name prefix as the stream using the retriever so that we don't need to assign a log level

Reply:
this ended up being a little more involved than I realized because the retriever's name was not properly evaluated https://github.com/airbytehq/airbyte/pull/18644/commits/26565d405e2a78694874495501248c9b05930234